Backpack/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt

```diff
@@ -59,23 +60,24 @@ data class CalendarParams(
   /**
    * Describes the selection behaviour
    */
-  enum class SelectionMode {
-
+  sealed interface SelectionMode {
     /**
      * No date can be selected
      */
-    Disabled,
+    object Disabled : SelectionMode
 
     /**
      * Only a single, non-disabled date can be selected.
      */
-    Single,
+    object Single : SelectionMode
 
     /**
      * Only a range can be selected.
      * The boundaries of range are always non-disabled dates, but there could be disabled dates within the range.
+     *
+     * @param allowSelectWholeMonth allows a whole month selection.
      */
-    Range,
+    data class Range(val allowSelectWholeMonth: Boolean = false) : SelectionMode
```

Comment:
Not sure if we discussed this before, but just curious about why we have `Range(val allowSelectWholeMonth: Boolean = false)` instead of having another type of the sealed interface, like `Month` (which is also a bit more consistent with the `CalendarSelection` class)?

Reply:
@sergiomarqmoura, Good point. However, this is the selection mode that describes the selection behaviour. @bvitaliyg and I agreed on this change instead of adding another sealed interface to minimise breaking changes.

Reply:
ah yes I think I understand the distinction now - I think that makes sense then. Follow up question is: should the label be part of the same range data class as it's only available for the range mode as well and linked to the boolean?

Reply:
> ah yes I think I understand the distinction now - I think that makes sense then. Follow up question is: should the label be part of the same range data class as it's only available for the range mode as well and linked to the boolean?

@marianeum, after reviewing your comment above -  adding the label to the data class for the range selection should be enough. We don't actually need the boolean since we can check if the label is available.